dexmaker-mockito-inline/src/main/java/com/android/dx/mockito/inline/MockMethodAdvice.java

```diff
@@ -86,6 +90,52 @@ private static Throwable hideRecursiveCall(Throwable throwable, int current,
         }
     }
 
+    private static final Map<String, String> PRIMITIVE_CLASS_TO_SIGNATURE =
+            Map.of(
+                    "byte", "B",
+                    "short", "S",
+                    "int", "I",
+                    "long", "J",
+                    "char", "C",
+                    "float", "F",
+                    "double", "D",
+                    "boolean", "Z");
+
+    /**
+     * Convert a type signature of an array to the corresponding class
+     *
+     * <p>It parse "foo[][][][]" into a jni signature "[[[[Lfoo;" and rely on Class.forName to do
+     * the conversion. Primivite type are converted using {@code PRIMITIVE_CLASS_TO_SIGNATURE}.
+     *
+     * @param argTypeName the type description (e.g: byte[])
+     * @return the class wrapped in Optional on success, or empty if the conversion failed
+     */
+    private static Optional<Class<?>> parseTypeName(String argTypeName) {
+        int index = argTypeName.indexOf("[");
+        if (index == -1) {
+            // not an Array type
+            return Optional.empty();
+        }
+
+        String typeName = argTypeName.substring(0, index);
+        String rest = argTypeName.substring(index, argTypeName.length());
+
+        if (!arrayPattern.matcher(rest).matches()) {
+            return Optional.empty();
+        }
+        int dimensionCount = (int) argTypeName.chars().filter(ch -> ch == '[').count();
+
+        String classSignature =
+                PRIMITIVE_CLASS_TO_SIGNATURE.getOrDefault(typeName, "L" + typeName + ";");
+
+        String fullTypeSignature = "[".repeat(dimensionCount) + classSignature;
```

Comment:
I see this fix changed in your AOSP commits. Do we need that fix here too?

Reply:
Sorry, I didn't update the commit here.

I did it know so this change is the same as the one I submitted in AOSP (works fine so far).

I remove usage of String.repeat, as it has been introduced in java 11 and I had some failures when the library got updated on old release test suite.
The safest option in my opinion was just to re-implement the repeat. That's a bit sad but I don't want to fight this battle and if one target failed there, that may have unforeseen consequences elsewhere